### PR TITLE
Fixed incorrect error on intersected number and string types

### LINKED ISSUE #782: Methods on types intersected with `number` cannot be used
Input:
```ts
type Vector = number & { normalize(): Vector };
(null! as Vector).normalize();
```

Current Result:
```
TSTL: Unsupported property on number: normalize
```

Expected Result:
No errors

### PR DESCRIPTION
**Pull request by Perryvw**
Fixes #782

### CODE REVIEW

File: test/unit/builtins/string.spec.ts [hunk 1]
```diff
@@ -1,22 +1,7 @@
...
-    const tsHeader = `
-        declare global {
-            interface String {
-                upper(): string;
-            }
-        }
-    `;
-
-    util.testExpression`"test".upper()`.setTsHeader(tsHeader).expectToEqual("TEST");
+    util.testExpression`"test".toUpperCase()`.expectToEqual("TEST");
```
Comment (ark120202) [inline test/unit/builtins/string.spec.ts]: Is that no longer supported?

File: test/unit/builtins/numbers.spec.ts [hunk 2]
```diff
@@ -67,3 +67,10 @@ test.each(cases)("isNaN(%p)", value => {
 test.each(cases)("isFinite(%p)", value => {
     util.testExpressionTemplate`isFinite(${value} as any)`.expectToMatchJsResult();
 });
+
+test("number intersected method", () => {
+    util.testFunction`
+        type Vector = number & { normalize(): Vector };
+        return ({ normalize: () => 3 } as Vector).normalize();
+        `.expectToMatchJsResult();
```
Comment (ark120202) [inline test/unit/builtins/numbers.spec.ts]:
```suggestion
    `.expectToMatchJsResult();
```

File: test/unit/builtins/string.spec.ts [hunk 3]
```diff
@@ -287,3 +272,10 @@ describe.each(["trim", "trimEnd", "trimRight", "trimStart", "trimLeft"])("string
         util.testExpression`${util.formatCode(testString)}.${trim}()`.expectToMatchJsResult();
     });
 });
+
+test("string intersected method", () => {
+    util.testFunction`
+        type Vector = string & { abc(): Vector };
+        return ({ abc: () => "a" } as Vector).abc();
+        `.expectToMatchJsResult();
```
Comment (ark120202) [inline test/unit/builtins/string.spec.ts]:
```suggestion
    `.expectToMatchJsResult();
```

(there are chances prettier would support it soon)

File: src/transformation/builtins/index.ts [hunk 4]
```diff
@@ -85,11 +92,11 @@ export function transformBuiltinCallExpression(
         }
     }
 
-    if (isStringType(context, ownerType)) {
+    if (isStringType(context, ownerType) && hasStandardLibrarySignature(context, node)) {
         return transformStringPrototypeCall(context, node);
     }
 
-    if (isNumberType(context, ownerType)) {
+    if (isNumberType(context, ownerType) && hasStandardLibrarySignature(context, node)) {
```
Comment (ark120202) [inline src/transformation/builtins/index.ts]: Could we use it for array (and maybe other types) as well?

File: src/LuaPrinter.ts [hunk 5]
```diff
@@ -742,11 +742,15 @@ export class LuaPrinter {
     public printMethodCallExpression(expression: lua.MethodCallExpression): SourceNode {
         const chunks = [];
 
-        const prefix = this.printExpression(expression.prefixExpression);
-
         const name = this.printIdentifier(expression.name);
 
-        chunks.push(prefix, ":", name, "(");
+        const prefix = this.printExpression(expression.prefixExpression);
```
Comment (ark120202) [inline src/LuaPrinter.ts]: How about something like that to remove parenthesizing duplication? 
```suggestion
        const prefix = this.printExpression(lua.isStringLiteral(expression.prefixExpression) ? expression.prefixExpression : lua.createParenthesizedExpression(expression.prefixExpression));;
```

Also, what about `"foo".bar`?

File: src/transformation/builtins/array.ts [hunk 6]
```diff
@@ -3,15 +3,13 @@ import * as lua from "../../LuaAST";
 import { TransformationContext } from "../context";
 import { UnsupportedProperty } from "../utils/errors";
 import { LuaLibFeature, transformLuaLibFunction } from "../utils/lualib";
-import { isExplicitArrayType } from "../utils/typescript";
 import { PropertyCallExpression, transformArguments } from "../visitors/call";
 
 export function transformArrayPrototypeCall(
     context: TransformationContext,
     node: PropertyCallExpression
 ): lua.CallExpression | undefined {
```
Comment (ark120202) [inline src/transformation/builtins/array.ts]:
```suggestion
): lua.CallExpression {
```

File: src/LuaPrinter.ts (hunk 5 above)
Comment (Perryvw) [reply src/LuaPrinter.ts]: Should not be affected by this PR, I'd prefer to keep that for the full parenthesis to printer change.

File: src/transformation/builtins/array.ts (hunk 6 above)
Comment (Perryvw) [reply src/transformation/builtins/array.ts]: That's not possible because the unsupported error is only thrown when isExplicitArrayType is true

**Pull request merged**